src/NLog/Targets/FileTarget.cs

```diff
@@ -938,11 +967,19 @@ private static void ArchiveFile(string fileName, string archiveFileName, bool en
             {
                 File.Move(fileName, archiveFileName);
             }
+
+            //We do this in case the archiveCallback is changed between the time we check the reference
+            // and when we make the callback.
+            var preventNullRefs = archiveCallback;
+            if (preventNullRefs != null)
+            {
+                preventNullRefs(archiveFileName);
+            }
```

Comment:
Code duplication. Should be call to `fileArchive_OnArchiveFileDeleted` / `fileArchive_OnFileArchived`?